tools/c7n_guardian/c7n_guardian/cli.py

```diff
@@ -301,42 +469,60 @@ def enable_region(master_info, accounts_config, executor, message, region):
         for f in as_completed(futures):
             a = futures[f]
             if f.exception():
-                log.error("Region:%s Error processing account:%s error:%s",
-                          region, a['name'], f.exception())
+                log.error(
+                    f"{region}\tError processing account:{a['name']} error:{f.exception()}"
+                )
                 continue
             if f.result():
-                log.info('Region:%s Enabled guard duty on account:%s',
-                         region, a['name'])
-    return members
+                log.info(f"{region}\tEnabled guard duty on account:{a['name']}")
+    return accounts_not_members
 
 
 def enable_account(account, master_account_id, region):
     member_session = get_session(
-        account.get('role'), 'c7n-guardian',
+        account.get('role'),
+        'c7n-guardian',
         profile=account.get('profile'),
-        region=region)
+        region=region,
+    )
     member_client = member_session.client('guardduty')
     m_detector_id = get_or_create_detector_id(member_client)
+    if not m_detector_id:
+        # Couldn't get a detector in this region; perhaps region not opted-in
+        log.info(
+            f"""{region}\tCouldn't get or create GuardDuty detector with role {account.get('role')}\
+- Perhaps the region isn't enabled?"""
+        )
+        return
     all_invitations = member_client.list_invitations().get('Invitations', [])
-    invitations = [
-        i for i in all_invitations
-        if i['AccountId'] == master_account_id]
+    invitations = [i for i in all_invitations if i['AccountId'] == master_account_id]
     invitations.sort(key=operator.itemgetter('InvitedAt'))
     if not invitations:
         log.warning(
-            "Region:%s No guard duty invitation found account:%s id:%s aid:%s",
-            region, account['name'], m_detector_id, account['account_id'])
+            f"""{region}\tNo guard duty invitation found Account Name:{account['name']} \
+AccountId: {account['account_id']} Detector ID:{m_detector_id}"""
+        )
         return
 
     member_client.accept_invitation(
         DetectorId=m_detector_id,
         InvitationId=invitations[-1]['InvitationId'],
-        MasterId=master_account_id)
+        MasterId=master_account_id,
+    )
     return True
 
 
 def get_or_create_detector_id(client):
-    detectors = client.list_detectors().get('DetectorIds')
+    try:
+        detectors = client.list_detectors().get('DetectorIds')
+    except ClientError as e:
+        # Occurs if the region is unavailable / not opted-in
+        # Regions available:        https://docs.aws.amazon.com/general/latest/gr/guardduty.html
+        # Regions requiring opt-in:
+        # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions
+        if e.response['Error']['Code'] == 'UnrecognizedClientException':
```

Comment:
is that error condition consistent? just wondering if it will change to the common which is auth error re opt in region being addressed.

Reply:
Unsure what you're asking sorry @kapilt ?
Yep that exception gets thrown both if invalid login, or region not opted-in
I tried running the script with invalid creds and it just fails with
`2020-05-21 13:54:00,825: c7n-guardian:ERROR us-west-2	Error processing: An error occurred (ExpiredToken) when calling the AssumeRole operation: The security token included in the request is expired`
which I think is probably ok?

Reply:
I think its okay.. but for context, there's two errors that are possible here

- guard duty is not available for the region ~ which results in a botocore.exceptions.ConnectTimeoutError after an automatic retry (looks like 3 times)

- botocore.exceptions.ClientError with code UnrecognizedClientException  .. for a region with guard duty that has not been opt'd in. I was referencing this is slightly unusual, as most services, return AuthFailure on the scenario, but for guard-duty.

There's an additional mismatch here to be aware of, namely that the master account that c7n-guardian is running from is used for expanding the symbolic `all` region, which may not have the same opt-in region list as the member accounts.

Reply:
True, things I didn't consider. Thanks very much for such a good review btw :)